refactor: Image cropping and optimization

mituldavid opened the pull request:
This PR makes the following changes to the image cropping and optimization feature introduced in https://github.com/frappe/frappe/pull/13835:
- Sets `allowed_file_types` for document images using MIME type instead of file extension
- Sets dialog backdrop as static during cropping
- Refactors file optimization API
- Removed `frappe.core.doctype.file.file.optimize_saved_image` in favour of `File.optimize_file` method
- Added rollback handler to re-write original file in case of rollback event
- Removes trailing white spaces

mituldavid committed:
refactor: Image cropping and optimization

gavindsouza reviewed (commented):
The best time to overwrite the file would probably be after this action has been committed. 

In the event of any failures, the document contents will be rolled back but the file will already be overwritten.

gavindsouza committed:
refactor: File.optimize_file

Simplified method logic

gavindsouza committed:
fix: Revert file on disk if rolledback

gavindsouza committed:
fix(File): Remove extra API endpoint

Got rid of frappe.core.doctype.file.file.optimize_saved_image in favour
of whitelisted Document Action File.optimize_file

gavindsouza committed:
Merge branch 'develop' into img-processing-changes

mituldavid committed:
fix: Flags not set on error during .save()

mituldavid committed:
refactor: Remove set_route after optimization

mituldavid committed:
test: Add tests for optimize_file

mituldavid committed:
test: Move tests for optimize_file to separate test case

mohsinalimat commented:
In latest version-14 **Image cropping** issue in web-form.


https://user-images.githubusercontent.com/973676/197758793-5bdadee3-e49b-4a6b-90b9-fab00ab662bc.mp4
